### PR TITLE
Convert bad_dtls_test for the new test framework

### PR DESCRIPTION
**Pull request by mattcaswell**
##### Checklist

- [x] tests are added or updated

### CODE REVIEW

File: test/bad_dtls_test.c [hunk 1]
```diff
@@ -182,7 +185,7 @@ static int validate_client_hello(BIO *wbio)
     /* Update handshake MAC for second ClientHello (with cookie) */
     if (cookie_found && !EVP_DigestUpdate(handshake_md, data + MAC_OFFSET,
                                           len - MAC_OFFSET))
-            printf("EVP_DigestUpdate() failed\n");
```
Comment (richsalz) [inline test/bad_dtls_test.c]: you don't want this to become TEST_info calls?

Comment (mattcaswell) [inline test/bad_dtls_test.c]: These calls should never fail really. The callers of `validate_client_hello` do wrap the return value in a `TEST` macro. If we add a TEST_info() for this then we should probably do all the "return 0" statements in this function (and probably the other similar functions). I thought that was a fairly low return on investment - but would do it if you insist.

Comment (richsalz) [inline test/bad_dtls_test.c]: As long as it was a conscious decision and you think we're not losing any practical diagnostic info, I'm fine with it.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

**Pull request closed without merging**